File: rust-core/tests/test_repos/simple_python_project/data_processor.py
class DataProcessor:
    """Class for processing data with multiple method calls"""
    
    def __init__(self):
        self.cache = {}
    
    def process_data(self, input_data):
        """Main processing method"""
        if input_data in self.cache:
            return self.cache[input_data]
        
        # Process the data through multiple steps
        cleaned_data = self._clean_input(input_data)
        validated_data = self._validate_data(cleaned_data)
        transformed_data = self._transform_data(validated_data)
        final_result = self._finalize_result(transformed_data)
        
        # Cache the result
        self.cache[input_data] = final_result
        return final_result
    
    def _clean_input(self, data):
        """Clean the input data"""
        if isinstance(data, str):
            return data.strip().lower()
        return str(data)
    
    def _validate_data(self, data):
        """Validate the cleaned data"""
        if not data:
            raise ValueError("Data cannot be empty")
        return data
    
    def _transform_data(self, data):
        """Transform the validated data"""
        # Apply multiple transformations
        step1 = self._step1_transform(data)
        step2 = self._step2_transform(step1)
        return step2
    
    def _step1_transform(self, data):
        """First transformation step"""
        return data.upper()
    
    def _step2_transform(self, data):
        """Second transformation step"""
        return f"PROCESSED_{data}"
    
    def _finalize_result(self, data):
        """Finalize the result"""
        return f"FINAL: {data}"
    
    def get_cache_stats(self):
        """Get cache statistics"""
        return {
            'size': len(self.cache),
            'keys': list(self.cache.keys())
        } 

File: rust-core/tests/test_repos/simple_js_project/dataProcessor.js
class DataProcessor {
    constructor() {
        this.cache = new Map();
    }
    
    processData(inputData) {
        // Check cache first
        if (this.cache.has(inputData)) {
            return this.cache.get(inputData);
        }
        
        // Process data through multiple steps
        const cleanedData = this._cleanInput(inputData);
        const validatedData = this._validateData(cleanedData);
        const transformedData = this._transformData(validatedData);
        const finalResult = this._finalizeResult(transformedData);
        
        // Cache the result
        this.cache.set(inputData, finalResult);
        return finalResult;
    }
    
    _cleanInput(data) {
        if (typeof data === 'string') {
            return data.trim().toLowerCase();
        }
        return String(data);
    }
    
    _validateData(data) {
        if (!data) {
            throw new Error('Data cannot be empty');
        }
        return data;
    }
    
    _transformData(data) {
        // Apply multiple transformations
        const step1 = this._step1Transform(data);
        const step2 = this._step2Transform(step1);
        return step2;
    }
    
    _step1Transform(data) {
        return data.toUpperCase();
    }
    
    _step2Transform(data) {
        return `PROCESSED_${data}`;
    }
    
    _finalizeResult(data) {
        return `FINAL: ${data}`;
    }
    
    getCacheStats() {
        return {
            size: this.cache.size,
            keys: Array.from(this.cache.keys())
        };
    }
    
    clearCache() {
        this.cache.clear();
    }
}

module.exports = DataProcessor; 